packages/shortest/src/tools/tool-registry.ts

```diff
@@ -0,0 +1,164 @@
+export type AnthropicToolType = z.infer<typeof anthropicToolTypeSchema>;
+
+// eslint-disable-next-line zod/require-zod-schema-types
+export type AnthropicModelFamily = "claude-3-5";
+
+// eslint-disable-next-line zod/require-zod-schema-types
+export type AnthropicToolVersion = "20241022";
+
+const ANTHROPIC_MODEL_TO_FAMILY: Record<AnthropicModel, AnthropicModelFamily> =
+  {
+    "claude-3-5-sonnet-latest": "claude-3-5",
+    "claude-3-5-sonnet-20241022": "claude-3-5",
+  };
+
+const ANTHROPIC_TOOL_VERSION_MAP: Record<
+  AnthropicModelFamily,
+  Record<AnthropicToolType, AnthropicToolVersion>
+> = {
+  "claude-3-5": {
+    computer: "20241022",
+    bash: "20241022",
+  },
+};
+
+export class ToolRegistry {
+  private tools: Map<string, ToolEntry> = new Map();
+  private log: Log;
+
+  constructor() {
+    this.log = getLogger();
+  }
+
+  public registerTool(key: string, entry: ToolEntry) {
+    if (this.tools.has(key)) {
+      throw new Error(`Tool with key '${key}' already registered`);
+    }
+    this.tools.set(key, entry);
+  }
+
+  public getTools(
+    provider: string,
+    model: AnthropicModel,
+    browserTool: BrowserTool,
+  ): Record<string, Tool> {
+    const selectedTools: Record<string, Tool> = {};
+
+    const providerTools = this.getProviderTools(provider, model, browserTool);
+    const customTools = this.getCustomTools(browserTool);
+
+    Object.assign(selectedTools, providerTools, customTools);
+
+    this.tools.forEach((entry, key) => {
+      if (!key.startsWith(`${provider}_`)) {
+        selectedTools[entry.name] = entry.factory(browserTool);
+      }
+    });
+
+    return selectedTools;
+  }
+
+  private getCustomTools(browserTool: BrowserTool): Record<string, Tool> {
+    const tools: Record<string, Tool> = {};
+
+    const customTools = Array.from(this.tools.values()).filter(
+      (entry) => entry.category === "custom",
+    );
+    customTools.forEach((entry) => {
+      tools[entry.name] = entry.factory(browserTool);
+    });
+
+    return tools;
+  }
+
+  private getProviderTools(
+    provider: string,
+    model: AnthropicModel,
+    browserTool: BrowserTool,
+  ): Record<string, Tool> {
+    const tools: Record<string, Tool> = {};
+
+    try {
+      const computerToolEntry = this.getProviderToolEntry(
+        provider,
+        model,
+        "computer",
+      );
+      tools[computerToolEntry.name] = computerToolEntry.factory(browserTool);
+    } catch (error) {
+      if (!(error instanceof ShortestError)) throw error;
+      this.log.trace("Computer tool not found for model, skipping", { model });
+    }
+
+    try {
+      const bashToolEntry = this.getProviderToolEntry(provider, model, "bash");
+      // @ts-ignore
+      // For some reason, it expects an argument, but it doesn't take any
+      tools["bash"] = bashToolEntry.factory();
+    } catch (error) {
+      if (!(error instanceof ShortestError)) throw error;
+      this.log.trace("Bash tool not found for model, skipping", { model });
```

Comment:
If a tool is not found (e.g. bash for 3.7 not yet available upstream), the package will still work without the tool.